test/e2e_tests/utils/testuser_adapter.go

```diff
@@ -0,0 +1,140 @@
+	ServiceAccountName       = "test-user"
+)
+
+// CreateTestUser creates a test user with the specified Kubernetes client and namespace.
+func CreateTestUser(k8sClient client.Client, namespace string) {
+	createTestUserServiceAccount(k8sClient, namespace)
+	createTestUserRoleAndRoleBinding(k8sClient, namespace)
+}
+
+// SwitchUser switches the Kubernetes client's user context to the test user if switchToTestUser is true.
+// If switchToTestUser is false, it reverts to the original context.
+func SwitchUser(k8sClient *client.Client, cfg *rest.Config, namespace string, scheme *runtime.Scheme, switchToTestUser bool) {
+	if switchToTestUser {
+		cfg.Impersonate = rest.ImpersonationConfig{
+			UserName: fmt.Sprintf(ServiceAccountNameFormat, namespace, ServiceAccountName),
+		}
+	} else {
+		cfg.Impersonate = rest.ImpersonationConfig{}
+	}
+
+	newClient, err := client.New(cfg, client.Options{Scheme: scheme})
+	Expect(err).NotTo(HaveOccurred())
+
+	// Update k8sClient's configuration with the new client's configuration
+	*k8sClient = newClient
+}
+
+// DeleteTestUser deletes the test user created in the specified namespace.
+func DeleteTestUser(k8sClient client.Client, namespace string) {
+	deleteTestUserRoleAndRoleBinding(k8sClient, namespace)
+	deleteTestUserServiceAccount(k8sClient, namespace)
+}
+
+// createTestUserServiceAccount creates a service account for the test user in the specified namespace.
+func createTestUserServiceAccount(k8sClient client.Client, namespace string) {
+	serviceAccount := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ServiceAccountName,
+			Namespace: namespace,
+		},
+	}
+
+	err := k8sClient.Create(context.Background(), serviceAccount)
+	Expect(err).To(SatisfyAny(BeNil(), WithTransform(errors.IsAlreadyExists, BeTrue())))
+}
+
+// createTestUserRoleAndRoleBinding creates a role and role binding for the test user in the specified namespace.
+func createTestUserRoleAndRoleBinding(k8sClient client.Client, namespace string) {
+	role := &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ServiceAccountName,
+			Namespace: namespace,
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{cappv1alpha1.GroupVersion.Group},
+				Resources: []string{"capps"},
+				Verbs:     []string{"get", "update"},
+			},
+		},
+	}
+
+	// Create or update the Role object
+	err := k8sClient.Create(context.Background(), role)
+	Expect(err).To(SatisfyAny(BeNil(), WithTransform(errors.IsAlreadyExists, BeTrue())))
+
+	roleBinding := &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ServiceAccountName,
+			Namespace: namespace,
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      ServiceAccountName,
+				Namespace: namespace,
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			Kind:     "Role",
+			Name:     ServiceAccountName,
+			APIGroup: "rbac.authorization.k8s.io",
+		},
+	}
+
+	err = k8sClient.Create(context.Background(), roleBinding)
```

Comment:
Probably will need to generalize the existing functions, but these creations should be moved to the `mocks` package https://github.com/dana-team/rcs-ocm-deployer/blob/main/test/e2e_tests/mocks/base.go